dvc/dependency/param.py

```diff
@@ -87,8 +88,15 @@ def read_params(self):
 
         with self.repo.tree.open(self.path_info, "r") as fobj:
             try:
-                config = yaml.safe_load(fobj)
-            except yaml.YAMLError as exc:
+                if self.path_info.suffix == ".yaml":
+                    config = yaml.safe_load(fobj)
```

Comment:
We use `yaml` to load json as well.

Reply:
Could just try checking `toml` suffix, and then fallback to `yaml.safe_load`.

Reply:
I just force-pushed a check for suffix (use `yaml.safe_load()` for json and yaml)

Reply:
Still this misses the `yml`. So, it's better to just:
```
if suffix == "toml":
    # toml load
else:
   # yaml load
```

Reply:
Is it actually safe for us to guess the file format from the suffix here? It seems like we'll potential to run into issues where users have params filenames with unexpected capitalization on case-sensitive filesystems, or where they are using `.yml` instead of `.yaml`.

It seems to me we should maybe just try to load as yaml (and json) first, then try toml if that fails, then error out?

Reply:
Agreed @pmrowla

Reply:
@pmrowla, if that's a concern, we can just do case-insensitive suffix checks. By doing the above, we'll be taxing users using TOML. Right now, it's just inside a params dependency, but for `dvc params diff/show`, it might get slow.

Reply:
Btw @dtrifiro, this probably won't make `dvc params diff/show` work. That'll require a change on:
https://github.com/iterative/dvc/blob/a4b3c6d872c06ef1259bb15bd0d69eaa4ccae7ff/dvc/repo/params/show.py#L29

Reply:
I added a loader map which is based on the lowercase file suffix, which defaults to `yaml.safe_load` for unknown suffixes. I also update `_read_params`

Reply:
I also agree with @pmrowla , it is better to just try to load and catch yaml formatting error than trying to judge by the extension.

Reply:
In the last force-push I ended up using another approach (default to `yaml.safe_load`, use `toml` if  lowercase suffix matches) because I felt a nested try/catch a bit awkward, especially if there are other loaders different from yaml/json and toml (not that I think there's much more, these 3 alone probably cover most of the use cases).

What is your preference?

Also, I saw that the Travis build fails on Windows and Mac (probably due to misconfiguration?) is there anything I need to do about that? Thanks

Reply:
@efiop, In case of unknown file format, sure, it could fallback to yaml; but, if the file is clearly a `toml` file, it should just try to load as toml and fail. 

This is what @dtrifiro is doing.

Reply:
Makes sense, let's keep it. :+1: